vault/resource_mongodbatlas_secret_role.go

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+var mongodbAtlasSecretBackendFromPathRegex = regexp.MustCompile("^(.+)/roles/.+$")
+
+func mongodbAtlasSecretRoleResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: mongodbAtlasSecretRoleCreateUpdate,
+		ReadContext:   ReadContextWrapper(mongodbAtlasSecretRoleRead),
+		UpdateContext: mongodbAtlasSecretRoleCreateUpdate,
+		DeleteContext: mongodbAtlasSecretRoleDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			consts.FieldMount: {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Path where MongoDB Atlas secret backend is mounted",
+				ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+			},
+			consts.FieldName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the role",
+			},
+			consts.FieldOrganizationID: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "ID for the organization to which the target API Key belongs",
+			},
+			consts.FieldProjectID: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "ID for the project to which the target API Key belongs",
+			},
+			consts.FieldRoles: {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "List of roles that the API Key needs to have",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			consts.FieldIPAddresses: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "IP address to be added to the whitelist for the API key",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			consts.FieldCIDRBlocks: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Whitelist entry in CIDR notation to be added for the API key",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			consts.FieldProjectRoles: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Roles assigned when an org API key is assigned to a project API key",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			consts.FieldTTL: {
+				Type:        schema.TypeString,
+				Computed:    true,
```

Comment:
I tested things out and looks like these are configurable, so we can update them to be `Optional`
```suggestion
				Optional:    true,
```